### PR TITLE
Upstreaming changes from disneystreaming/smithy-language-server

### PR DESCRIPTION
**Pull request by daddykotex**
I've rebased our commits on top of your main branch.

I tried to remove all references to Disney Streaming Services fork (scala, mill, publishing under our org, etc), but let me know if you find some left overs.

### CODE REVIEW

File: src/main/java/software/amazon/smithy/lsp/SmithyInterface.java [hunk 1]
```diff
@@ -26,15 +34,64 @@ private SmithyInterface() {
...
-      return Either.forRight(Model.assembler().discoverModels().addImport(path.getAbsolutePath()).assemble());
+      Model.Builder builder = Model.builder();
+
+      List<URL> urls = externalJars.stream().map(SmithyInterface::fileToUrl).collect(Collectors.toList());
+      URL[] urlArray = urls.toArray(new URL[urls.size()]);
+
+      if (urlArray.length > 0) {
+        // Loading the model just from upstream dependencies, in isolation, and adding
+        // all its
+        // shapes to the builder.
```
Comment (srchase) [inline src/main/java/software/amazon/smithy/lsp/SmithyInterface.java:56]: nit: formatting

File: src/main/java/software/amazon/smithy/lsp/ext/Completions.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (srchase) [inline src/main/java/software/amazon/smithy/lsp/ext/Completions.java:2]: nit: Wrong copyright year on the new files.

**Review by srchase — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ## Smithy Language Server
+
 [![Build Status](https://github.com/awslabs/smithy-language-server/workflows/ci/badge.svg)](https://github.com/awslabs/smithy-language-server/actions/workflows/ci.yml)
 
 A [Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
```
[the remark this answers is not in the document]
Comment (daddykotex) [reply README.md:5]: the presentation on GitHub hides it but it still there, we did not touch it. the diff on this file is just whitespace, I'll restore your README.md

[the remark this answers is not in the document]
Comment (srchase) [reply README.md]: Thanks!

**Review by srchase — APPROVED** [review]

File: src/main/java/software/amazon/smithy/lsp/ext/Completions.java (hunk 2 above)
Comment (daddykotex) [reply src/main/java/software/amazon/smithy/lsp/ext/Completions.java:2]: To be clear, do you want all copyright notices to be updated to 2022, or you'd like files that we introduced to have our (disney streaming) Copyright notice?

**Review by srchase — APPROVED** [review]

**Pull request merged**